test/sql/generated_columns/virtual/alter_table.test

```diff
@@ -0,0 +1,47 @@
+# name: test/sql/generated_columns/virtual/alter_table.test
+# description: Test generated columns added through ALTER TABLE query
+# group: [virtual]
+
+statement ok
+CREATE TABLE unit (price INTEGER, amount_sold INTEGER);
+
+statement ok
+INSERT INTO unit VALUES (5,4)
+
+# Function doesnt exist, neither does the referenced column
+statement error
+ALTER TABLE unit ADD COLUMN total_profit BOOLEAN GENERATED ALWAYS AS(non_existant_function() * price) VIRTUAL;
```

Comment:
Generated columns should not require an explicit type specification, no? The type can be determined from the function itself.